app/src/main/java/com/celzero/bravedns/ui/UniversalFirewallFragment.kt

```diff
@@ -204,14 +205,10 @@ class UniversalFirewallFragment : Fragment(R.layout.universal_fragement_containe
         }
     }
 
-    private fun openCustomIpDialog() {
-        val themeId = Themes.getCurrentTheme(isDarkThemeOn(), persistentState.theme)
-        val customDialog = CustomIpDialog(requireActivity(), customIpViewModel, themeId)
-        customDialog.show()
-    }
-
-    private fun isDarkThemeOn(): Boolean {
-        return resources.configuration.uiMode and Configuration.UI_MODE_NIGHT_MASK == Configuration.UI_MODE_NIGHT_YES
+    private fun openCustomIpScreen() {
+        val intent = Intent(requireContext(), CustomIpActivity::class.java)
+        intent.flags = Intent.FLAG_ACTIVITY_RESET_TASK_IF_NEEDED
```

Comment:
why is this flag used? Pls add a comment.

Reply:
Added comment.